pkg/kubectl/resource_filter.go

```diff
@@ -0,0 +1,110 @@
+package kubectl
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+type FilterFunc func(runtime.Object) (bool, error)
+
+type FilterOptions struct {
+	filterMap    map[reflect.Type]reflect.Value
+	options      PrintOptions
+	hiddenObjNum int
+}
+
+func NewResourceFilter(opts *PrintOptions) *FilterOptions {
+	filterOpts := &FilterOptions{
+		filterMap: make(map[reflect.Type]reflect.Value),
+		options:   *opts,
+	}
+
+	filterOpts.addDefaultHandlers()
+	return filterOpts
+}
+
+func (f *FilterOptions) addDefaultHandlers() {
+	f.Handler(filterPods)
+}
+
+// Handler adds a filter handler to a FilterOptions instance.
+// See validateFilterFunc for required method signature.
+func (f *FilterOptions) Handler(handlerFn interface{}) error {
```

Comment:
Don't really like `Handler` as a name.  `AddFilter`?

Also, why are you taking an interface and then doing validation, why not just explicitly take a `FilterFunc`?